src/main/java/heronarts/lx/Tempo.java

```diff
@@ -191,16 +194,38 @@ public Tempo(LX lx) {
     addParameter("trigger", this.trigger);
     addParameter("enabled", this.enabled);
     addModulator("nudge", this.nudge);
+
+    // set initial minBpm, maxBpm
+    minBpm = DEFAULT_MIN_BPM;
+    maxBpm = DEFAULT_MAX_BPM;
   }
 
   private static final String PATH_BEAT = "beat";
   private static final String PATH_SET_BPM = "setBPM";
+  
+  public boolean setBpmRange(double min, double max) {
+    if (min <= 0.0 || min >= max) {
+      // do not set to invalid range of BPMs
+      return false;
+    }
+    
+    minBpm = min;
+    maxBpm = max;
+    return true;
+  }
+
+  public boolean isValidBpm(double bpm) {
+    return bpm >= minBpm && bpm <= maxBpm;
+  }
 
   @Override
   public boolean handleOscMessage(OscMessage message, String[] parts, int index) {
     if (parts[index].equals(PATH_SET_BPM)) {
       if (message.size() > 0) {
-        this.bpm.setValue(message.getFloat());
+        float newBpm = message.getFloat();
+        if (isValidBpm(newBpm)) {
+          this.bpm.setValue(message.getFloat());
```

Comment:
This should be `this.bpm.setValue(newBpm);`

`message.getFloat()` will try to pull ANOTHER argument off the OSC message, after you've already called `message.getFloat()` once.

I might throw an error log message in the case that the BPM is not valid... it's probably good to know when a client is sending invalid data. Understand that it may not always be possible to prevent with 3rd-party OSC software so this change seems sensible in general, but invalid messages are def something to try to get out of the system.

Reply:
OH wow. welp that explains a lot of the issues I was having with OSC in LX.....